pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -434,6 +434,11 @@ func translateRoute(
 		Match:    translateRouteMatch(node, virtualService, match),
 		Metadata: util.BuildConfigInfoMetadata(virtualService.Meta),
 	}
+
+	if match != nil && match.StatPrefix != "" {
+		out.StatPrefix = match.StatPrefix
```

Comment:
Why made StatPrefix in  `HTTPMatchRequest`? This API is to specify matching conditions.

Reply:
Because we want metrics per match (path) condition. If service exposes multiple paths, we want stats per route. This is mainly useful for Ingress Gateway.